tests/test_tools.py

```diff
@@ -728,3 +730,51 @@ def test_group_member_ids(input, expected):
     actual = group_member_ids(input, "bg")
     assert actual == expected
 
+p1 = {
+    "_id": "scopatz",
+    "aka": [
+        "Scopatz",
+        "Scopatz, A",
+        "Scopatz, A.",
+        "Scopatz, A M",
+        "Anthony Michael Scopatz",
+    ],
+    "name": "Anthony Scopatz",
+}
+p2 = {
+    "_id": "abc",
+    "aka": [
+        "A. BC",
+        "BC, A",
+        "Anthony BC",
+    ],
+    "name": "Anthony Bill Chris",
+}
+
+@pytest.mark.parametrize(
+    "input, expected",
+    [
+        (([p1, p2], ["aka", "name", "_id"],
+                           "Anth", False),[p1,p2]),
+        (([p1, p2], ["aka", "name", "_id"],
+                           "scopatz, a", True),[]),
+        (([p1, p2], ["aka", "name", "_id"],
+                           "scopatz, a", False),[p1]),
+        (([p1, p2], ["aka", "name", "_id"],
+                           "ill", False),[p2]),          
+    ],
+)
+def test_fragment_retrieval(input, expected):
+    assert(fragment_retrieval(input[0],input[1],input[2],input[3]) == expected)
```

Comment:
to make it easier to read, could you do:
```
    assert(fragment_retrieval(input[0], input[1], input[2], case_sensitive=input[3]) == expected)
```
 Also, your code is not pepe8 compliant.  If you remind me at group meeting I can show you how to quickly fix that in pycharm.  We can also use flake8.  I can show that too.